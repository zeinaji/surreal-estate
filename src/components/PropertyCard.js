import React from "react";
import "../styles/PropertyCard.css";
import logo from "../property.png";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faBed, faBath } from "@fortawesome/free-solid-svg-icons";
import { faEnvelope } from "@fortawesome/free-regular-svg-icons";

const PropertyCard = ({
  title,
  type,
  bathrooms,
  bedrooms,
  price,
  city,
  email,
}) => {
  return (
    <div className="property-card">
      <img className="logo" src={logo} alt="logo" />
      <div className="title">{title}</div>
      <div className="type-city">{`${type} - ${city}`}</div>
      <div className="bathrooms">
        <FontAwesomeIcon icon={faBath} />
        {bathrooms}
      </div>
      <div className="bedrooms">
        <FontAwesomeIcon icon={faBed} />
        {bedrooms}
      </div>
      <div className="price">{`Â£${price}`}</div>
      <div className="email">
        <a className="email-link" href={`mailto:${email}`}>
          <FontAwesomeIcon className="envelope" icon={faEnvelope} />
          Email
        </a>
      </div>

      {/* <form method="post" action={`mailto:${email}`}>
        <input type="submit" value={logo} />
  </form> */}
    </div>
  );
};

export default PropertyCard;
