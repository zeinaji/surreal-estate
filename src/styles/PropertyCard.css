.property-card {
  width: 200px;
  height: 220px;
  margin: 10px;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;
  border: 2px solid grey;
}

.email {
  height: 30px;
  width: 70px;
  text-align: center;
  background-color: grey;
}

.email-link {
  margin: auto;
  color: white;
  display: inline-block;
  margin-top: 4px;
}

.email-link:hover {
  background: white;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.envelope {
  margin-right: 3px;
}
