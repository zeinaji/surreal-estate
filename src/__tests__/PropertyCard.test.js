import React from "react";
import PropertyCard from "../components/PropertyCard";
import { render } from "@testing-library/react";

describe("PropertyCard", () => {
  it("renders", () => {
    const { asFragment } = render(<PropertyCard />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with the correct props", () => {
    const property = {
      title: "2 Bedroom Flat",
      type: "Flat",
      city: "Manchester",
      bathrooms: 2,
      bedrooms: 3,
      price: 1000.0,
      email: "example@example.com",
    };
    // const { title, type, city, bathrooms, bedrooms, price, email } = property;

    const { getByText, getByAltText } = render(
      <PropertyCard
        {...property}
        /* title={title}
        type={type}
        city={city}
        bathrooms={bathrooms}
        bedrooms={bedrooms}
        price={price}
    email={email} */
      />
    );

    const logo = getByAltText("logo");
    expect(logo).toBeInTheDocument();

    const titleElement = getByText("2 Bedroom Flat");
    expect(titleElement).toHaveClass("title");

    const typeElement = getByText(/Flat - /);
    expect(typeElement).toHaveClass("type-city");

    const cityElement = getByText(/Manchester/);
    expect(cityElement).toHaveClass("type-city");

    const bathroomsElement = getByText("2");
    expect(bathroomsElement).toHaveClass("bathrooms");

    const bedroomsElement = getByText("3");
    expect(bedroomsElement).toHaveClass("bedrooms");

    const priceElement = getByText(/1000/);
    expect(priceElement).toHaveClass("price");

    expect(document.querySelector("a").getAttribute("href")).toBe(
      "mailto:example@example.com"
    );
  });
});
