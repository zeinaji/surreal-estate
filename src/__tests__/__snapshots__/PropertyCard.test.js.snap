// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <img
      alt="logo"
      class="logo"
      src="house.png"
    />
    <div
      class="title"
    />
    <div
      class="type-city"
    >
      undefined - undefined
    </div>
    <div
      class="bathrooms"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bath fa-w-16 "
        data-icon="bath"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32,384a95.4,95.4,0,0,0,32,71.09V496a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V480H384v16a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V455.09A95.4,95.4,0,0,0,480,384V336H32ZM496,256H80V69.25a21.26,21.26,0,0,1,36.28-15l19.27,19.26c-13.13,29.88-7.61,59.11,8.62,79.73l-.17.17A16,16,0,0,0,144,176l11.31,11.31a16,16,0,0,0,22.63,0L283.31,81.94a16,16,0,0,0,0-22.63L272,48a16,16,0,0,0-22.62,0l-.17.17c-20.62-16.23-49.83-21.75-79.73-8.62L150.22,20.28A69.25,69.25,0,0,0,32,69.25V256H16A16,16,0,0,0,0,272v16a16,16,0,0,0,16,16H496a16,16,0,0,0,16-16V272A16,16,0,0,0,496,256Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="bedrooms"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bed fa-w-20 "
        data-icon="bed"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="price"
    >
      Â£undefined
    </div>
    <div
      class="email"
    >
      <a
        class="email-link"
        href="mailto:undefined"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-envelope fa-w-16 envelope"
          data-icon="envelope"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
            fill="currentColor"
          />
        </svg>
        Email
      </a>
    </div>
  </div>
</DocumentFragment>
`;
